overlord/hookstate/hookmgr.go

```diff
@@ -401,7 +413,11 @@ func (m *HookManager) runHook(context *Context, snapst *snapstate.SnapState, hoo
 	if f := m.hijacked(hooksup.Hook, hooksup.Snap); f != nil {
 		err = f(context)
 	} else if hookExists {
-		output, err = runHook(context, tomb)
+		if hookNotUsableError != nil {
```

Comment:
Why do we do this check here instead of right where the !errors.As() happens?

Reply:
This change was reverted due to proposal https://github.com/snapcore/snapd/pull/12701#discussion_r1171303190